src/AuthorizationService.php

```diff
@@ -50,34 +51,48 @@ public function __construct(ResolverInterface $resolver)
     /**
      * @inheritDoc
      */
-    public function can(?IdentityInterface $user, string $action, $resource)
+    public function can(?IdentityInterface $user, string $action, $resource): ResultInterface
```

Comment:
If the service is consistently returning a Result how does `Identity::can()` return booleans?

Reply:
I forgot to update Identity class :) Guess we don't have any tests which cover integration between both service and identity.

Reply:
So there was a test but it only tested with the service returning Result hence nothing failed :).

Reply:
That's unfortunate

Reply:
My question was more around the breakage and complexity spillage that applications are exposed to. By always returning a Result we force everyone to deal with the most complex scenario (a Result) when the primary usage is returning booleans.

Reply:
It would add one extra method call for checks but it would make your app code future proof. Boolean results might suffice when starting out but let's say down the line you want to add / update a policy which returns `Result` instance, since you want to provide a message with reason for failure.

You would then have to go about updating all checks in your app to check whether return type of `can()` is boolean or `Result` instance.

I am open to alternate suggestions to avoid such code refactoring.

Reply:
Could we add a `canResult()` method to the IdentityDecorator & AuthorizationService? That would let people opt-in to 'expert mode' of getting results back.

Reply:
We could, and change `can()` to always return boolean?

Reply:
Yeah. It is a breaking change from 1.x but a smaller one than having can() always return Result objects.